R/tm_g_heat_bygrade.R

```diff
@@ -352,62 +421,21 @@ srv_g_heatmap_bygrade <- function(id,
     })
 
     output_q <- reactive({
-      iv_len <- shinyvalidate::InputValidator$new()
-      anno_var <- input$anno_var
-      iv_len$add_rule("anno_var", function(x) if (length(x) > 2) "Please include no more than 2 annotation variables.")
-      iv_len$enable()
-      validate(need(iv_len$is_valid(), "Misspecification error: please observe red flags in the encodings."))
-      validate(need(iv$is_valid(), "Misspecification error: please observe red flags in the encodings."))
-
       ADSL <- data[[sl_dataname]]() # nolint
       ADEX <- data[[ex_dataname]]() # nolint
       ADAE <- data[[ae_dataname]]() # nolint
-
-      validate(need(nrow(ADSL) > 0, "Please select at least one subject"))
-      validate(need(
-        input$ongo_var %in% names(ADEX),
-        paste("Study Ongoing Status must be a variable in", ex_dataname, sep = " ")
-      ))
-
-      validate(need(
-        checkmate::test_logical(ADEX[[input$ongo_var]], min.len = 1, any.missing = FALSE),
-        "Study Ongoing Status must be a logical variable"
-      ))
-
-      validate(need(
-        all(anno_var %in% names(ADSL)),
-        paste("Please only select annotation variable(s) in", sl_dataname, sep = " ")
-      ))
-
-      validate(need(
-        !(input$id_var %in% anno_var),
-        paste("Please de-select", input$id_var, "in annotation variable(s)", sep = " ")
-      ))
-
       if (isTRUE(input$plot_cm)) {
         ADCM <- data[[cm_dataname]]() # nolint
-        validate(
-          need(
-            input$conmed_var %in% names(ADCM),
-            paste("Please select a Conmed Variable in", cm_dataname, sep = " ")
-          )
-        )
-        validate(need(
-          is.factor(ADCM[[input$conmed_var]]),
-          "Conmed Variable should be a factor"
-        ))
-        validate(need(
-          all(input$conmed_level %in% levels(ADCM[[input$conmed_var]])),
-          "Updating Conmed Levels"
-        ))
       }
 
+      teal::validate_has_data(ADSL, min_nrow = 0, msg = sprintf("%s contains no data", sl_dataname))
+
+      teal::validate_inputs(iv(), iv_cm())
```

Comment:
it's up to you whether `isTRUE(input$plot_cm)` should be here to conditionally add `iv_cm` to the validator. Alternatively, it could be in `iv_cm` in the way I suggested there

Reply:
I don't understand. `iv_cm` is not "added to the validator", it is a separate object.

Reply:
Please ignore this suggestion, I was wrong by thinking that whole `iv_cm()` has a rule on `isTRUE(input$plot_cm)`. It's fine now

Reply:
It does, actually, that is why there is a separate validator object.
I think we are in agreement now.